rtt_ros2_params/src/rtt_ros2_params.cpp

```diff
@@ -0,0 +1,296 @@
+  if (get_ros2_node(rosnode)) {
+    if (!rosnode->has_parameter(param_name)) {
+      // Create new parameter
+      RTT::log(RTT::Warning) <<
+        "The parameter did not exist, creating one" <<
+        RTT::endlog();
+      rosnode->declare_parameter(param_name);
+    }
+
+    // Update the parameter
+    try {
+      const auto result = rosnode->set_parameter(rclcpp::Parameter(param_name, paramvalue));
+      if (!result.successful) {
+        RTT::log(RTT::Error) <<
+          "The parameter " << param_name <<
+          " couldn't be set with reason: " << result.reason << RTT::endlog();
+        return false;
+      }
+    } catch (std::exception & e) {
+      RTT::log(RTT::Error) <<
+        "The parameter failed to be set, reason: " <<
+        e.what() << RTT::endlog();
+    }
+  } else {
+    RTT::log(RTT::Error) <<
+      "The ROS2 node doesn't exist, no parameter can be set. Import rtt_ros2_node first and use "
+      "the \"ros2-node\" service." <<
+      RTT::endlog();
+    return false;
+  }
+  return true;
+}
+
+bool Params::loadProperty(const std::string param_name, const std::string property_name)
+{
+  RTT::Logger::In in(getName());
+
+  rclcpp::ParameterValue paramvalue = getParameter(param_name);
+  if (rclcpp::PARAMETER_NOT_SET == paramvalue.get_type()) {
+    RTT::log(RTT::Error) <<
+      "The parameter \"" << param_name << "\" couldn't be retrieved" <<
+      RTT::endlog();
+    return false;
+  }
+
+  if (nullptr != getOwner()->properties()->getProperty(property_name)) {
+    RTT::log(RTT::Debug) <<
+      getOwner()->provides()->getName() <<
+      " has a property " << property_name << RTT::endlog();
+  } else {
+    RTT::log(RTT::Info) <<
+      getOwner()->provides()->getName() <<
+      " has NO property " << property_name << RTT::endlog();
+    for (const auto property : getOwner()->properties()->getProperties() ) {
+      RTT::log(RTT::Debug) <<
+        "Available property: " << property->getName() <<
+        RTT::endlog();
+    }
+  }
+
+  auto prop = getOwner()->properties()->getProperty(property_name);
+  if (nullptr == prop) {
+    // Try to find it among orphan parameters (previously loaded, without a component interface)
+    if (orphan_params_.find(property_name) == orphan_params_.end()) {
+      // When totally new, then create an orphan property, i.e. owned by the ros2-params service
+      orphan_params_[property_name] = paramvalue;
+      this->addProperty(property_name, orphan_params_[property_name])
+      .doc("Property loaded from ROS2 param: " + param_name);
+      prop = getProperty(property_name);
+    } else {
+      orphan_params_[property_name] = paramvalue;
+      prop = getProperty(property_name);
+    }
+    return true;
+  } else {
+    RTT::Property<rclcpp::ParameterValue> prop_paramvalue = prop;
+
+    // The property exists on the Owner's interface
+    RTT::log(RTT::Debug) <<
+      "The property " << property_name << " existed" <<
+      RTT::endlog();
```

Comment:
```suggestion
    // The property exists on the Owner's interface
    RTT::log(RTT::Debug) <<
      "The property " << property_name << " existed" <<
      RTT::endlog();

    RTT::Property<rclcpp::ParameterValue> prop_paramvalue = prop;
```

The comment/debug message refers to the `else` case, not `prop_paramvalue`.